src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from "react";
+import * as CheckboxPrimitive from "@radix-ui/react-checkbox";
+import { cn } from "../../utilities";
+
+const CheckboxRoot = CheckboxPrimitive.Root;
+
+const CheckboxIndicator = CheckboxPrimitive.Indicator;
+
+const Checkbox = React.forwardRef<
+	React.ElementRef<typeof CheckboxPrimitive.Root>,
+	React.ComponentPropsWithoutRef<typeof CheckboxPrimitive.Root>
+>(({ className, ...props }, ref) => (
+	<CheckboxPrimitive.Root
+		ref={ref}
+		className={cn(
+			[
+				"flex h-4 w-4 shrink-0 items-center justify-center rounded-[4px] border border-theme-border-bold bg-theme-surface-primary",
+				"hover:ring-4 hover:ring-theme-surface-tertiary",
+				"focus:outline-none focus:ring-4 focus:ring-theme-surface-tertiary",
+				"data-[state=checked]:border-theme-surface-strong data-[state=checked]:bg-theme-surface-strong data-[state=checked]:active:border-primary-200 data-[state=checked]:active:bg-primary-200",
+				"active:bg-theme-surface-tertiary",
+				"disabled:pointer-events-none disabled:border-neutral-300 disabled:data-[state=checked]:border-neutral-300 disabled:data-[state=checked]:bg-neutral-300"
+			],
+			className
+		)}
+		{...props}
+	>
+		<CheckboxPrimitive.Indicator className={cn("flex items-center justify-center text-current")}>
+			<svg
```

Comment:
I needed to hard embed the svg here, in case we need a more flexible usecase in the future, I also exported the base primitives `CheckboxRoot` and `CheckboxIndicator`

Reply:
I think for the moment it is enough embedding the svg but we have to find a way to "componentise" the icons. 🤔